Source/MediaInfo/Audio/File_Usac.cpp

```diff
@@ -1803,16 +2252,485 @@ void File_Usac::UsacFrame(size_t BitsNotIncluded)
+                if(max_sfb1>max_sfb)
+                    max_sfb_ste=max_sfb1;
+            TESTELSE_SB_END();
+            Get_S1(2, ms_mask_present,                          "ms_mask_present");
+            if (ms_mask_present==1)
+            {
+                for (int8u group=0; group<num_window_groups; group++)
+                    for (int8u sfb=0; sfb<max_sfb_ste; sfb++)
+                        Skip_SB(                                "ms_used[g][sfb]");
+            }
+            else if (ms_mask_present==3 && C.stereoConfigIndex==0)
+                cplxPredData(max_sfb_ste, usacIndependencyFlag);
+        TEST_SB_END();
+        if (C.tw_mdct)
+        {
+            TEST_SB_GET(C.common_tw,                            "common_tw");
+                twData();
+            TEST_SB_END();
+        }
+
+        if (tns_active)
+        {
+            bool common_tns=false;
+            if (C.common_window)
+                Get_SB(common_tns,                              "common_tns");
+            Skip_SB(                                            "tns_on_lr");
+
+            if (common_tns)
+            {
+                //TODO: tnsData();
+                tns_data_present0=false;
+                tns_data_present1=false;
+            }
+            else
+            {
+                TESTELSE_SB_SKIP(                               "tns_present_both");
+                    tns_data_present0=true;
+                    tns_data_present1=true;
+                TESTELSE_SB_ELSE(                               "tns_present_both");
+                    Get_SB(tns_data_present1,                   "tns_data_present[1]");
+                    tns_data_present0=!tns_data_present1;
+                TESTELSE_SB_END();
+            }
+        }
+        else
+        {
+            tns_data_present0=0;
+            tns_data_present1=0;
+        }
+    }
+    else
+    {
+        C.common_window=false;
+        C.common_tw=false;
+    }
+
+    Element_End0();
+}
+
+//---------------------------------------------------------------------------
+void File_Usac::UsacCoreCoderData(size_t nrChannels, bool usacIndependencyFlag)
+{
+    Element_Begin1("UsacCoreCoderData");
+
+    bool coreModes[2];
+    bool tnsDataPresent[2];
+
+    for (size_t ch=0; ch<nrChannels; ch++)
+        Get_SB(coreModes[ch],                                   "core_mode");
+
+    if (nrChannels==2)
+        StereoCoreToolInfo(tnsDataPresent[0], tnsDataPresent[1], coreModes[0], coreModes[1], usacIndependencyFlag);
+
+    for (size_t ch=0; ch<nrChannels; ch++)
+    {
+        if (coreModes[ch])
+        {
+            //TODO: lpd_channel_stream(indepFlag);
+        }
+        else
+        {
+            if (nrChannels==1 || coreModes[0]!=coreModes[1])
+                Get_SB(tnsDataPresent[ch],                      "tns_data_present");
+
+            fdChannelStream(ch, C.common_window, C.common_tw, tnsDataPresent[ch], usacIndependencyFlag);

```

Comment:
Add:

```

    if (!Aac_sampling_frequency[C.sampling_frequency_index])
        ch=nrChannels;
```

(for smooth stop of parsing in case of unknown frequency index)